doc/onboarding.md

```diff
@@ -0,0 +1,156 @@
+      * improvement doesn't have to come all at once
+    * set time limits and expectations:
+      * the collaborators are very distributed so it is unlikely that they will be looking at stuff the same time as you are.
+      * before merging code: give folks at least one working day to respond: "If no one objects, tomorrow at <time> I'll merge this in."
+        * technically, 48 hours for non-trivial changes, and 72 hours on weekends.
+        * set reminders
+      * check in on the code every once in a while (set reminders!)
+      * if a PR is abandoned, check if they'd mind if you took it over (especially if its just nits left)
+    * what belongs in node:
+      * opinions vary, but I find the following helpful:
+      * if node itself needs it, then it belongs in node
+        * that is to say, url is there because of http, freelist is there because of http, et al
+      * also, things that cannot be done outside of core, or only with great pain (i.e. async-wrap)
+
+
+### process for getting code in:
+  * the collaborator guide is a great resource: https://github.com/nodejs/node/blob/master/COLLABORATOR_GUIDE.md#technical-howto
+  * no one (not even TSC or CTC) pushes directly to master without review
+    * best practice: commit often, out to your github fork (origin), open a PR
+    * make sure to spend time on the description:
+     * every moment you spend writing a good description quarters the amount of time it takes to understand your code.
+    * prefer to only squash at the *end* of your work
+    * one "LGTM" is usually sufficient, except for semver-major changes
+      * the more the better
+      * semver-major (breaking) changes must be reviewed in some form by the CTC
+    * you have the power to LGTM another collaborator or TSC / CTC members' work
+    * **make sure to run the PR through CI before merging!** (Except for documentation PRs)
+  * once code is ready to go in:
+    * `git checkout master && git status`
+    * if the working directory is clean
+    * `git remote update -p && git merge --ff-only upstream/master`
+    * To land a PR
+    * if it all looks good, `curl -L 'url-of-pr.patch' | git am`
+    * `git rebase -i upstream/master`
+    * squash into logical commits
+    * `./configure && make -j8 test` (`-j8` builds iojs in parallel with 8 threads. adjust to the number of cores (or processor-level threads) your processor has for best results.)
+    * commits should follow `subsystem[,subsystem]: small description\n\nbig description\n\n<metadata>`
+    * first line 50 columns, all others 72
+    * add metadata:
+      * `Fixes: <url>`
+      * `Reviewed-By: human <email> `
+        * Easiest to use `git log` then do a search (`/QUERY` + `enter` (+ `n` as much as you need to) in vim)
+      * `PR-URL: <original-full-pr-url>`
+    * `git push upstream master`
+    * close the original PR with "Landed in `<commit hash>`".
```

Comment:
If it's your own PR, you can do `git push mine +HEAD:branch && git push origin master && git push mine :branch`.  The PR will show as merged instead of closed (purple vs red.)